### PR TITLE
feat(release): publish JSII Python Artifacts to AWS CodeArtifact

### PR DESCRIPTION
**Pull request by timothymathison**
Makes pypi publisher job support targeting AWS CodeArtifact. Re-uses the `CodeArtifactOptions` already added to support publishing of NPM packages to CodeArtifact.

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/release/publisher.ts [hunk 1]
```diff
@@ -501,25 +501,74 @@ export class Publisher extends Component {
...
+          uses: "aws-actions/configure-aws-credentials@v4",
+          with: {
+            "role-to-assume": roleToAssume,
+            "aws-region": region,
+          },
+        });
+      }
+      prePublishSteps.push({
+        name: "Generate CodeArtifact Token",
+        run: `echo "TWINE_PASSWORD=$(aws codeartifact get-authorization-token --domain ${domain} --domain-owner ${account} --region ${region} --query authorizationToken --output text)" >> $GITHUB_ENV`,
```
Comment (timothymathison) [inline src/release/publisher.ts:539]: Unfortunately publib does not natively support codeartifact for python like it does for npm. So had to add this step.

File: src/release/publisher.ts [hunk 2]
```diff
@@ -1190,6 +1246,28 @@ export function isAwsCodeArtifactRegistry(registryUrl: string | undefined) {
...
+  readonly account?: string;
+  readonly region?: string;
+}
+
+/**
+ * Parses info about code artifact domain from given AWS code artifact url
+ * @param url Of code artifact domain
+ * @returns domain, account, and region of code artifact domain
+ */
+export function awsCodeArtifactInfoFromUrl(url?: string): AwsCodeArtifactInfo {
```
Comment (mrgrain) [inline src/release/publisher.ts:1263]: No need for this to be exported.
```suggestion
function awsCodeArtifactInfoFromUrl(url?: string): AwsCodeArtifactInfo {
```

File: src/release/publisher.ts [hunk 3]
```diff
@@ -1190,6 +1246,28 @@ export function isAwsCodeArtifactRegistry(registryUrl: string | undefined) {
   return registryUrl && AWS_CODEARTIFACT_REGISTRY_REGEX.test(registryUrl);
 }
 
+/**
+ * Info extracted from AWS CodeArtifact URL
+ */
+export interface AwsCodeArtifactInfo {
```
Comment (mrgrain) [inline src/release/publisher.ts:1252]: No need for this to be exported and part of the public API.

```suggestion
interface AwsCodeArtifactInfo {
```

**Review by mrgrain — CHANGES_REQUESTED** [review]
Couple of minor changes. Note that removing the export will also require an update the the generated docs again.

**Review by mrgrain — APPROVED** [review]
Nice one! This was long overdue

**Pull request merged**